File: leetcode-submissions/0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String ans = strs[0];
        for (int i = 1; i < strs.length; i++) {
            ans = longestCommonPrefixBetweenTwostrings(ans, strs[i]);
            if (ans == "") return ans;
        }
        return ans;
    }

    public String longestCommonPrefixBetweenTwostrings(String a, String b) {
        char[] aChars = a.toCharArray();
        char[] bChars = b.toCharArray();
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < aChars.length && i < bChars.length; i++)  {
            if (aChars[i] != bChars[i]) {
                return sb.toString();
            } else {
                sb.append(aChars[i]);
            }
        }
        return sb.toString();
    }
}

File: leetcode-submissions/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Get the smallest length of the string
        # Create a variable for prefix string
        # For loop for the range of the minimum number
        # If the string is present in all strings ass it to the res else break
        # Return the string
        length = float("inf")
        for i, n in enumerate(strs):
            length = min(length, len(n))

        print(length)
        
        if length < 1:
            return ""

        res = ""

        for i in range(length):
            char = strs[0][i]
            end = False
            for j in range(len(strs)):
                if strs[j][i] != char:
                    end = True
                    break

            if not end:
                res += strs[0][i]
            if end:
                break

        return res
